miasm2/jitter/llvmconvert.py

```diff
@@ -483,25 +503,25 @@ def set_ret(self, var):
             var_casted = var
         self.builder.ret(var_casted)
 
-    @staticmethod
-    def canonize_label_name(label):
+    def canonize_label_name(self, label):
         """Canonize @label names to a common form.
         @label: str or asmlabel instance"""
         if isinstance(label, str):
             return label
-        if m2_asmblock.expr_is_label(label):
-            label = label.name
-        if isinstance(label, m2_asmblock.AsmLabel):
-            if label.offset is None:
-                return "label_%s" % label.name
-            else:
-                return "label_%X" % label.offset
+        if not isinstance(label, LocKey):
+            raise ValueError("label must either be str or LocKey")
+
+        name = self.llvm_context.ir_arch.symbol_pool.loc_key_to_name(label)
+        offset = self.llvm_context.ir_arch.symbol_pool.loc_key_to_offset(label)
+        print 'XX', label, name
+        if offset is None:
+            return "%s" % name
         else:
-            raise ValueError("label must either be str or asmlabel")
+            return "label_off_%X" % offset
 
-    def get_basic_bloc_by_label(self, label):
+    def get_basic_bloc_by_loc_key(self, loc_key):
```

Comment:
If the name has to changed, a `bloc` -> `block` could be great too

Reply:
done